From 7876465d20f7ddff48161734fea58acc887efb19 Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Thu, 22 Jan 2026 15:32:23 -0700
Subject: [PATCH] chore: Allow for CMake policy_max of 4.2

* CMake v4 is out as of 2025 and extending the policy_max makes builds more
  accessible across CMake releases.
* As this allows for CMake policies up to 4.2, this also addresses warnings
  for CMake policy CMP0167, which was added in CMake v3.30.
   - c.f. https://cmake.org/cmake/help/latest/policy/CMP0167.html
---
 CMakeLists.txt               | 2 +-
 tests/install/CMakeLists.txt | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 83fe7c5..2dda7db 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
 
 # Set range of valid CMake versions to build the project.
-cmake_minimum_required(VERSION 3.1...3.20)
+cmake_minimum_required(VERSION 3.1...4.2)
 
 if(${CMAKE_VERSION} VERSION_LESS 3.12)
     cmake_policy(VERSION ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION})
diff --git a/tests/install/CMakeLists.txt b/tests/install/CMakeLists.txt
index 01362ab..a47b498 100644
--- a/tests/install/CMakeLists.txt
+++ b/tests/install/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.12...3.20)
+cmake_minimum_required(VERSION 3.12...4.2)
 project(TestLwtnn)
 
 set( CMAKE_CXX_STANDARD 11 CACHE STRING
-- 
2.52.0

